### PR TITLE
python3Packages.wfuzz: fix build on Python 3.12

### LINKED ISSUE #326902: wfuzz error - no module named 'imp'
### Describe the bug
wfuzz installation returns the following error:
```
...
  File "/nix/store/p93gaplc5svnpvpik8k0aprrgfa9xql2-python3.12-wfuzz-3.1.0/lib/python3.12/site-packages/wfuzz/__init__.py", line 55, in <module>
    from .options import FuzzSession
  File "/nix/store/p93gaplc5svnpvpik8k0aprrgfa9xql2-python3.12-wfuzz-3.1.0/lib/python3.12/site-packages/wfuzz/options.py", line 6, in <module>
    from .facade import (
  File "/nix/store/p93gaplc5svnpvpik8k0aprrgfa9xql2-python3.12-wfuzz-3.1.0/lib/python3.12/site-packages/wfuzz/facade.py", line 5, in <module>
    from .externals.moduleman.loader import DirLoader
  File "/nix/store/p93gaplc5svnpvpik8k0aprrgfa9xql2-python3.12-wfuzz-3.1.0/lib/python3.12/site-packages/wfuzz/externals/moduleman/loader.py", line 3, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'
error: builder for '/nix/store/f9453pjz3ljmqp5xcqhp5f9ixsz37dk1-python3.12-wfuzz-3.1.0.drv' failed with exit code 1;
```

### Steps To Reproduce
1. `nix-shell -p wfuzz`

### Notify maintainers
@Pamplemousse 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.37, Athena OS, 24.11 (Vicuna), 24.11pre652179.7e7c39ea35c5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"home-manager, nixos, nixpkgs"`
 - nixpkgs: `/home/athena/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghpzin**
## Description of changes

- add patch from upstream PR fixing imp module uses
(https://github.com/xmendez/wfuzz/pull/365)
- disable test depending on imp module
- replace `distutils` with `setuptools._distutils`

Fixes #326902 (`wfuzz` build failure)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ghpzin** [comment]
Result of `nixpkgs-review pr 329970` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python311Packages.wfuzz</li>
    <li>python311Packages.wfuzz.dist</li>
    <li>wfuzz (python312Packages.wfuzz)</li>
    <li>wfuzz.dist (python312Packages.wfuzz.dist)</li>
    <li>wordlists</li>
  </ul>
</details>

**Comment by ghpzin** [comment]
Result of `nixpkgs-review pr 329970` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
    <li>python311Packages.wfuzz</li>
    <li>python311Packages.wfuzz.dist</li>
    <li>wfuzz (python312Packages.wfuzz)</li>
    <li>wfuzz.dist (python312Packages.wfuzz.dist)</li>
    <li>wordlists</li>
  </ul>
</details>

File: pkgs/development/python-modules/wfuzz/default.nix [hunk 1]
```diff
@@ -29,9 +30,22 @@ buildPythonPackage rec {
...
+    })
+  ];
+
   postPatch = ''
     substituteInPlace setup.py \
       --replace "pyparsing>=2.4*" "pyparsing>=2.4"
+
+    # fix distutils use for Python 3.12
+    substituteInPlace src/wfuzz/plugin_api/base.py \
+      --replace "from distutils import util" "from setuptools._distutils import util"
```
Comment (pbsds) [inline pkgs/development/python-modules/wfuzz/default.nix:48]:
```suggestion
      --replace-fail "from distutils import util" "from setuptools._distutils import util"
```

Comment (ghpzin) [reply pkgs/development/python-modules/wfuzz/default.nix]: done

**Comment by ghpzin** [comment]
Result of `nixpkgs-review pr 329970` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python311Packages.wfuzz</li>
    <li>python311Packages.wfuzz.dist</li>
    <li>wfuzz (python312Packages.wfuzz)</li>
    <li>wfuzz.dist (python312Packages.wfuzz.dist)</li>
    <li>wordlists</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**